pytorch_lightning/plugins/native_amp.py

```diff
@@ -56,9 +56,17 @@ def backward(self, closure_loss, optimizer, opt_idx, *args, **kwargs):
 
         return closure_loss
 
-    def clip_gradients(self, grad_clip_val: Union[int, float], optimizer: Optimizer, norm_type: float):
+    def clip_gradients(self,
+                       optimizer: Optimizer,
+                       grad_clip_val: Union[int, float],
+                       gradient_clip_algorithm: str,
+                       norm_type: Union[float, int]):
         model = self.trainer.get_model()
         torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=grad_clip_val, norm_type=norm_type)
+        if gradient_clip_algorithm == 'value':
```

Comment:
@tchaton @dhkim0225 this should work as-is with shardedDDP, this time the sharding is not an issue since the gradients are clipped on an individual basis. Per rank, some of the `model.parameters()` will just have no gradient since they'll be on another rank, but clip_grad_value will just skip them. Let me know if you have issues around that, this is just an educated guess

Reply:
Ah, you are right.
Since `clip_by_value` is not affected by model sharding, normal torch.nn.utils.grad_value_() method should work.

Reply:
@blefaudeux Thanks for your comment, you are right. I'll close fairscale issue.

Reply:
@dhkim0225 In PR description, the following comment should be removed :-)

> "However, using clip_grad_by_val with sharded_ddp plugin is not implemented for now, since fairscale also do not have clip_grad_value feature."

Reply:
removed!